data.py

```diff
@@ -201,24 +243,24 @@ def load_sensor_data(self, data: str) -> None:
             sensor_value = sensor["value"]
 
             if sensor_name == "gps":
-                self.data["gps"] = 1
-                self.data["gps_speed"] = float(sensor_value["speed"])
+                self.data["gps"].update(1)
+                self.data["gps_speed"].update(float(sensor_value["speed"]))
             elif sensor_name == "reedVelocity":
-                self.data["reed_velocity"] = float(sensor_value)
-            elif sensor_name in self.data_types:
-                cast_func = self.data_types[sensor_name]
-                self.data[sensor_name] = cast_func(sensor_value)
+                self.data["reed_velocity"].update(float(sensor_value))
```

Comment:
One quick addition: Can you duplicate these two lines to also update `reed_distance` when we get data on `reedDistance`? Was going to make my own PR but I realise that it would cause a conflict with yours so probably just faster that you add it in here.

Reply:
Actually, these could be done inside the `else` statement right? Since this is now just the same thing

Reply:
Not quite the same thing! (annoyingly)

In `self.data`, the keys are names of the V2 data fields which use snake_case, but in V3 we're using CamelCase. (I'm not entirely sure why that was changed... It was though). So, for these multi-word data names, we need to manually map the snake case into camel case.

Reply:
Ohhh I see the issue now. Haha that's so strange